profiling/src/profile/mod.rs

```diff
@@ -561,6 +633,49 @@ impl Profile {
             .get(&local_root_span_id)
             .map(i64::clone))
     }
+
+    fn upscale_values(&self, values: &[i64], labels: &[Label]) -> anyhow::Result<Vec<i64>> {
+        let mut new_values = values.to_vec();
+
+        if !self.upscaling_rules.is_empty() {
+            let mut values_to_update: Vec<usize> = vec![0; self.sample_types.len()];
+
+            // get bylabel rules first (if any)
+            let mut group_of_rules = labels
+                .iter()
+                .filter_map(|label| self.upscaling_rules.get(&(label.key, label.str)))
+                .collect::<Vec<&Vec<UpscalingRules>>>();
+
+            // get byvalue rules if any
+            if let Some(byvalue_rules) = self.upscaling_rules.get(&(0, 0)) {
+                group_of_rules.push(byvalue_rules);
+            }
+
+            // check for collision(s)
+            group_of_rules.iter().for_each(|rules| {
+                rules.iter().for_each(|rule| {
+                    rule.values_offset
+                        .iter()
+                        .for_each(|offset| values_to_update[*offset] += 1)
+                })
+            });
+
+            anyhow::ensure!(
+                values_to_update.iter().all(|v| *v < 2),
+                "Multiple rules modifying the same offset for this sample"
+            );
+
+            group_of_rules.iter().for_each(|rules| {
+                rules.iter().for_each(|rule| {
+                    rule.values_offset
+                        .iter()
+                        .for_each(|offset| new_values[*offset] *= rule.ratio as i64)
```

Comment:
This first converts `ratio` to a `i64` and then multiply the value by it, I would expect to first convert value to f64 then scale the value then convert it back to i64 (should the conversion do a round to nearest, floor or round towards zero ?).